File: core/src/main/java/xtdb/vector/IValueReader.java
package xtdb.vector;

import clojure.lang.Keyword;

import java.nio.ByteBuffer;

public interface IValueReader {
    Keyword getLeg();

    boolean isNull();
    boolean isAbsent();

    boolean readBoolean();
    byte readByte();
    short readShort();
    int readInt();
    long readLong();

    float readFloat();
    double readDouble();

    ByteBuffer readBytes();
    Object readObject();
}

File: core/src/main/kotlin/xtdb/vector/IValueReader.kt
package xtdb.vector

import clojure.lang.Keyword
import java.nio.ByteBuffer

interface IValueReader {
    val leg: Keyword?

    val isNull: Boolean

    fun readBoolean(): Boolean
    fun readByte(): Byte
    fun readShort(): Short
    fun readInt(): Int
    fun readLong(): Long

    fun readFloat(): Float
    fun readDouble(): Double

    fun readBytes(): ByteBuffer
    fun readObject(): Any?
}
